syntax = "proto3";

package comm;

service fileTransfer {
  // Uploads a file
  rpc UploadFile(stream UploadFileRequest) returns (StringResponse) {}
  // Downloads a file
  rpc DownloadFile(MetaData) returns (stream FileResponse) {}
}


message StringResponse {
  string message = 1;
}

message FileResponse {
  bytes chunk_data = 1;
}

message MetaData {
  string filename = 1;
  string extension = 2;
}

message UploadFileRequest {
  oneof request {
    MetaData metadata = 1;
    bytes chunk_data = 2;
  }
}